Assets/Scripts/UI/Tooltip.cs

```diff
@@ -0,0 +1,224 @@
+                    tooltipStyle.fontSize = tooltipStyleShadow.fontSize = Mathf.RoundToInt(tooltipSize - (cameraDistance / 3));
+                }
+                GUI.Label(new Rect(Event.current.mousePosition.x + tooltipPosition.x - 40, Event.current.mousePosition.y + tooltipPosition.y, 100f, 20f), image, tooltipStyle);
+                GUI.Label(new Rect(Event.current.mousePosition.x + tooltipPosition.x - tooltipShadowPosition.x, Event.current.mousePosition.y + tooltipPosition.y - tooltipShadowPosition.y, 100f, 20f), currentText, tooltipStyleShadow);
+                GUI.Label(new Rect(Event.current.mousePosition.x + tooltipPosition.x, Event.current.mousePosition.y + tooltipPosition.y, 100f, 20f), currentText, tooltipStyle);
+                if (tooltipUIPanel != null)
+                {
+                    tooltipUIPanel.transform.localPosition = new Vector3(Event.current.mousePosition.x + tooltipPosition.x - Screen.width / 2f, -Event.current.mousePosition.y + tooltipPosition.y + Screen.height / 2f, 0);
+                }
+            }
+            // Display of text/tooltip that fixed to the object
+            else if (!hideTooltip && fixedToTheObject && over)
+            {
+                positionToScreen = Camera.main.WorldToScreenPoint(transform.position);
+                cameraDistance = Vector3.Distance(Camera.main.transform.position, this.transform.position);
+                if (textResized)
+                {
+                    tooltipStyle.fontSize = tooltipStyleShadow.fontSize = Mathf.RoundToInt(tooltipSize - (cameraDistance / 3));
+                }
+                //GUI.DrawTexture(new Rect((positionToScreen.x + tooltipPosition.x) - 40f, -positionToScreen.y + Screen.height + tooltipPosition.y / (cameraDistance / 10), 100f, 20f), image, ScaleMode.ScaleToFit);
+                GUI.Label(new Rect(positionToScreen.x + tooltipPosition.x - 40, -positionToScreen.y + Screen.height + tooltipPosition.y / (cameraDistance / 10), 100f, 20f), image, tooltipStyle);
+                GUI.Label(new Rect(positionToScreen.x + tooltipPosition.x - tooltipShadowPosition.x, -positionToScreen.y + Screen.height + tooltipPosition.y / (cameraDistance / 10) - tooltipShadowPosition.y, 100f, 20f), currentText, tooltipStyleShadow);
+                GUI.Label(new Rect(positionToScreen.x + tooltipPosition.x, -positionToScreen.y + Screen.height + tooltipPosition.y / (cameraDistance / 10), 100f, 20f), currentText, tooltipStyle);
+                if (tooltipUIPanel != null)
+                {
+                    tooltipUIPanel.transform.localPosition = new Vector3(positionToScreen.x + tooltipPosition.x - Screen.width / 2f, positionToScreen.y - Screen.height / 2f + tooltipPosition.y / (cameraDistance / 10), positionToScreen.z);
+                }
+            }
+        }
+
+        // Show tooltip and focus color of this object
+        public void beginTooltip()
```

Comment:
method names

Reply:
Yeah couldn't think of a better name since it's not really showing the Tooltip it just setting up. I guess i will named it setTooltip and resetTooltip then ?